aws/resource_aws_efs_access_point.go

```diff
@@ -0,0 +1,404 @@
+		Update: resourceAwsEfsAccessPointUpdate,
+		Delete: resourceAwsEfsAccessPointDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"file_system_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"file_system_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"owner_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"posix_user": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"gid": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"uid": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: true,
+						},
+						"secondary_gids": {
+							Type:     schema.TypeSet,
+							Elem:     &schema.Schema{Type: schema.TypeInt},
+							Set:      schema.HashInt,
+							Optional: true,
+							ForceNew: true,
+						},
+					},
+				},
+			},
+			"root_directory": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"path": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							Computed: true,
+						},
+						"creation_info": {
+							Type:     schema.TypeList,
+							Optional: true,
+							ForceNew: true,
+							MaxItems: 1,
+							Computed: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"owner_gid": {
+										Type:     schema.TypeInt,
+										Required: true,
+										ForceNew: true,
+									},
+									"owner_uid": {
+										Type:     schema.TypeInt,
+										Required: true,
+										ForceNew: true,
+									},
+									"permissions": {
+										Type:     schema.TypeString,
+										Required: true,
+										ForceNew: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsEfsAccessPointCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).efsconn
+
+	fsId := d.Get("file_system_id").(string)
+
+	input := efs.CreateAccessPointInput{
+		FileSystemId: aws.String(fsId),
+		Tags:         keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().EfsTags(),
+	}
+
+	if v, ok := d.GetOk("posix_user"); ok {
+		input.PosixUser = expandEfsAccessPointPosixUser(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("root_directory"); ok {
+		input.RootDirectory = expandEfsAccessPointRootDirectory(v.([]interface{}))
+	}
+
+	log.Printf("[DEBUG] Creating EFS Access Point: %#v", input)
+
+	ap, err := conn.CreateAccessPoint(&input)
+	if err != nil {
+		return err
```

Comment:
Nit: We should return error context here for operators and code maintainers:

```suggestion
		return fmt.Errorf("error creating EFS Access Point for File System (%s): %w", fsId, err)
```